crates/cranelift/src/func_environ.rs

```diff
@@ -2646,7 +2647,14 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
     ) -> WasmResult<()> {
         // If the `vmruntime_limits_ptr` variable will get used then we initialize
         // it here.
-        if self.tunables.consume_fuel || self.tunables.epoch_interruption {
+        if true || self.tunables.consume_fuel || self.tunables.epoch_interruption {
```

Comment:
Idea: we should add a field to `tunables`, e.g. `tunables.resume` which we set whenever we enter `translate_resume`.

Reply:
Unfortunately, this needs to have run before we execute `translate_resume` for the first time, so that we an use the runtime limits `Variable`.

Reply:
Set it in cranelift then? Or does it need to be set when enter function is run?

Reply:
The function we are looking at (`before_translate_function`) is called to perform some setup before `func_translator::parse_function_body` starts actual translation of a particular function. In the current version of this PR, this setup must run before `translate_resume` does. So the only option would be to detect inside `parse_function_body` that there is a `resume` in the function body, and then indicate to `before_translate_function` that we want `declare_vmruntime_limits_ptr` to be executed. But `parse_function_body` isn't designed to iterate over the instructions in the function twice.

Long story short: That's all rather annoying. Given that it has no runtime overhead, I decided to leave it as is for now.

Reply:
OK, thanks for clarifying. It would be good to open an issue about this "problem".

Reply:
#105